packages/sitecore-jss-react/src/components/Placeholder.tsx

```diff
@@ -73,15 +123,27 @@ class PlaceholderComponent extends PlaceholderCommon<PlaceholderComponentProps>
       renderingData,
       this.props.name
     );
+
+    const isEmptyPlaceholder = placeholderData.every(
+      (rendering: ComponentRendering | HtmlElementRendering) => isRawRendering(rendering)
+    );
+
     const components = this.getComponentsForRenderingData(placeholderData);
 
-    if (
-      this.props.renderEmpty &&
-      placeholderData.every((rendering: ComponentRendering | HtmlElementRendering) =>
-        isRawRendering(rendering)
-      )
-    ) {
-      return this.props.renderEmpty(components);
+    if (isEmptyPlaceholder) {
+      if (
+        typeof window !== 'undefined' &&
+        HorizonEditor.isActive() &&
+        !this.placeholderRef.current
+      ) {
+        this.collectHorizonEmptyPlaceholders();
+      }
+
+      return (
+        <div ref={this.placeholderRef}>
```

Comment:
Same as comment on test - any way we can only add the div (for ref purposes, etc) when in Horizon editing context? Ideally, this doesn't get added under "normal" page rendering (provides backwards compatibility for non-Horizon projects as well).

Reply:
@ambrauer We don't have ability to check that we are in Horizon or Exp Editor, when we are on the server side.
So If here I will add check for Horizon, in this case I will have mismatch between server markup and rendered markup on the client side. In this case it will not work as expected

Reply:
@illiakovalenko I see your point. My concern is that this extra div could potentially affect customer's JS/CSS. Can we use a fragment instead (pretty sure you mentioned that's a "no", but just checking)? If extra div is the only way, then definitely something we'd have to note on release notes.

Reply:
Unfortunately, we can't use `ref` with `Fragments`